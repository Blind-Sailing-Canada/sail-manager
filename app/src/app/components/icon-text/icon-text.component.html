<div class="icon-text">
  <span class="icon-icon">
    <mat-icon inline="true">{{icon}}</mat-icon>
  </span>
  <span>{{text}}</span>
</div>
