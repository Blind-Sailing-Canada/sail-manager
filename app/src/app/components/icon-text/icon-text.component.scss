.icon-text {
  display: inline-flex;
}

.icon-icon {
  display: inline-flex;
  align-items: center;
  margin-right: 5px;
}