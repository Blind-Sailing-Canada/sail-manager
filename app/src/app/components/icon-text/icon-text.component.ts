import {
  Component,
  Input,
} from '@angular/core';

@Component({
  selector: 'app-icon-text',
  templateUrl: './icon-text.component.html',
  styleUrls: ['./icon-text.component.css']
})
export class IconTextComponent {

  @Input() icon: string;
  @Input() text: string;

}
