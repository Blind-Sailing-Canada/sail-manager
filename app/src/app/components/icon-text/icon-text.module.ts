import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IconTextComponent } from './icon-text.component';
import { AngularMaterialModule } from '../../angular-material/angular-material.module';

@NgModule({
  declarations: [
    IconTextComponent,
  ],
  exports: [
    IconTextComponent,
  ],
  imports: [
    AngularMaterialModule,
    CommonModule,
  ]
})
export class IconTextModule { }
