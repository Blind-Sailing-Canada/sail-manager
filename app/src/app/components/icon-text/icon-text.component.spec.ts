import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { IconTextModule } from './icon-text.module';
import { Component } from '@angular/core';

@Component({
  template: `
    <app-icon-text
    text="FOOBAR"></app-icon-text>`
})
class TestHostComponent {

}

describe('Icon-Text Component', () => {
  let fixture: ComponentFixture<TestHostComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        IconTextModule
      ],
      declarations: [TestHostComponent],
      providers: [],
    }).compileComponents()
      .then(() => {
        fixture = TestBed.createComponent(TestHostComponent);
        fixture.detectChanges();
      })
      .catch(error => console.error(error));
  }));
  it('should display text', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
