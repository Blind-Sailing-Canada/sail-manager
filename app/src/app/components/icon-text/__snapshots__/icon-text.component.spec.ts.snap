// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon-Text Component should display text 1`] = `
<div
  id="root0"
>
  <app-icon-text
    text="FOOBAR"
  >
    <div
      class="icon-text"
    >
      <span
        class="icon-icon"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons mat-icon-inline mat-icon-no-color"
          inline="true"
          role="img"
        >
          
        </mat-icon>
      </span>
      <span>
        FOOBAR
      </span>
    </div>
  </app-icon-text>
</div>
`;
